lib/linguist/languages.yml

```
@@ -3690,6 +3690,13 @@ Lean:
   tm_scope: source.lean
   ace_mode: text
   language_id: 197
+Lean 4:
+  type: programming
```

As previously suggested:
```suggestion
  type: programming
  group: Lean
```

I also recommend taking this opportunity to add a colour; the same for both should be good.

---

I'm going to punt on choosing the color since that requires an almost literal bikeshedding discussion with the community, and I'd rather not hold up this PR on that. I promise to start that discussion once this is merged, and open a follow-up PR!

---

There's no need for community buy-in when first adding a colour. The requirement is only there for changing a colour as some people are really sensitive about it 😁 

I was going to suggest picking a colour from the logo or site but the [Lean lang website](https://lean-lang.org/) is totally devoid of colour so I guess the default grey `#cccccc` is quite apt 😆 

We'll leave it as-is. No need to engage with the community unless you really want to.